--- a/src/socket_lookup.h
+++ b/src/socket_lookup.h
@@ -368,7 +368,7 @@ int ipv6_find_hdr(const struct sk_buff *skb, unsigned int *offset,
 	return nexthdr;
 }
 
-#  if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,37)
+#  if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,32)
 extern struct sock *__udp6_lib_lookup(struct net *net,
 				      struct in6_addr *saddr, __be16 sport,
 				      struct in6_addr *daddr, __be16 dport,
--- a/src/socket_notify.c
+++ b/src/socket_notify.c
@@ -26,7 +26,7 @@
 
 #include "socket_notify.h"
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,33)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,32)
 #define DECLARE_CREATE_HOOK(NAME) int NAME(struct net *net, struct socket *sock, int protocol, int kern) 
 #define CALL_CREATE_HOOK(NAME) NAME(net, sock, protocol, kern)
 #else
